### ISSUE TITLE
getActionForPathAndParams returns undefined params if input params is {}

### ISSUE THREAD

**Issue by leethree** [react-navigation/react-navigation#3029]
### Current Behavior

After #2976, `StackRouter.getActionForPathAndParams` will return an action without `params` when `inputParams` is empty object `{}`. 

_e.g._
```js
router.getActionForPathAndParams('articles', {});
```
Returns: 
```js
{
  routeName: "ArticleList",
  type: "Navigation/NAVIGATE"
}
```

This causes errors for screens that expect `params` to be non-null.

### Expected Behavior
I'm not sure what is the correct behavior here. But before `v1.0.0-beta.20`, the behavior was:

```js
router.getActionForPathAndParams('articles', {});
```
Returns: 
```js
{
  routeName: "ArticleList",
  type: "Navigation/NAVIGATE",
  params: {}
}
```

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 1.0.0-beta.21
| react-native     | 0.50.3
| node             | 8.5.0
| yarn      | 1.3.2

**Comment by eriveltonelias** [comment]
Hi @leethree! 

I'll close this because it's related to an old `react-navigation` library. But if the problem persists, please open a new issue, please.
